### PR TITLE
DropDown menu

### PR DESCRIPTION
**Pull request by Sebastien-Ahkrin**
Parts of https://github.com/zakodium-oss/analysis-ui-components/issues/260

### CODE REVIEW

File: src/components/menu/MenuItems.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    </ClassNames>
+  );
+}
+
+function Item<T>(props: ItemProps<T>) {
+  const { option, onSelect } = props;
+  const isDivider = option.type === 'divider';
+
+  return (
+    <Menu.Item disabled={!isDivider && option.disabled}>
```
Comment (stropitek) [inline src/components/menu/MenuItems.tsx:91]: The divider should not render a `Menu.Item`. `Menu.Item` is for elements that can be active and selected with the keyboard (so only options of type `option`)

File: src/components/menu/MenuItems.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  );
+}
+
+function ItemOption<T>(props: ItemOptionProps<T>) {
+  const { onSelect, option, active } = props;
+  const { disabled } = option;
+
+  return (
+    <ClassNames>
+      {({ css }) => (
```
Comment (stropitek) [inline src/components/menu/MenuItems.tsx:111]: There is a simpler way to pass css. See for example https://github.com/zakodium-oss/analysis-ui-components/blob/main/src/components/value-renderers/Color.tsx

The first line of the file is important!

File: src/components/menu/MenuItems.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        <div
+          onClick={() => onSelect(option)}
+          className={css([
+            {
+              ...styles.item,
+              display: 'flex',
```
Comment (stropitek) [inline src/components/menu/MenuItems.tsx:117]: did you not use grid (was suggested in the issue) for a reason?

File: stories/components/menuitems.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import { Menu } from '@headlessui/react';
+import { useMemo } from 'react';
+import { FaMeteor } from 'react-icons/fa';
+
+import { MenuItems, MenuOptions } from '@/components/menu/MenuItems';
+
+export default {
+  title: 'Components / MenuItems',
```
Comment (stropitek) [inline stories/components/menuitems.stories.tsx:8]: You can move these stories and the "Dropdown" stories to a common "Components / DropdownMenu" folder.

File: src/components/menu/MenuItems.tsx (hunk 3 above)
Comment (Sebastien-Ahkrin) [reply src/components/menu/MenuItems.tsx:117]: no, just a personnal preference, i will use grid on the next push :)

Comment (Sebastien-Ahkrin) [reply src/components/menu/MenuItems.tsx]: what, how can i reproduce this bug ?

**Comment by stropitek** [comment]
image

The labels should be aligned vertically, like in [this screenshot](https://user-images.githubusercontent.com/2352663/197483598-331a0702-0b3f-476f-8f1a-e96d4a6b819e.png)

File: src/components/menu/MenuItems.tsx [hunk 5]
```diff
@@ -107,8 +107,8 @@ function ItemOption<T>(props: ItemOptionProps<T>) {
       css={css([
         {
           ...styles.item,
-          display: 'flex',
-          flexDirection: 'row',
+          display: 'grid',
```
Comment (targos) [inline src/components/menu/MenuItems.tsx:110]: It's not the option that should be a grid. It's the entire menu.

**Comment by lpatiny** [comment]
When you open the drop down menu and then click somewhere else I would expect the menu closes: https://deploy-preview-283--analysis-ui-components.netlify.app/stories/?story=components--dropdownmenu--dropdown

Also the context menu seems to crash: https://deploy-preview-283--analysis-ui-components.netlify.app/stories/?story=components--dropdownmenu--context-dropdown

**Comment by Sebastien-Ahkrin** [comment]
> When you open the drop down menu and then click somewhere else I would expect the menu closes: https://deploy-preview-283--analysis-ui-components.netlify.app/stories/?story=components--dropdownmenu--dropdown
> 
> Also the context menu seems to crash: https://deploy-preview-283--analysis-ui-components.netlify.app/stories/?story=components--dropdownmenu--context-dropdown

yeah sorry @lpatiny i'm still working on it, i mark the Pull request as draft sorry !

**Comment by lpatiny** [comment]
I'm eager that we replace in NMRium the current drop down by this new one that looks soo much better !!!

I have a specific case for which I would like a different icon color. Something like:

image

Does the current implementation allows this ?

**Comment by targos** [comment]
> Does the current implementation allows this ?

Yes, the icon can be any React element

**Pull request merged**